### PR TITLE
Add missing reset for ping clocks to avoid mistakenly sending GOAWAY frames due to 'too_many_pings'

### LINKED ISSUE #23146: Client receives GOAWAY frames when it receives lot of trailer-only responses
Internal bug - b/156897072

@apolcyn noticed a bug where clients received GOAWAY frames in an environment where the Client sends a lot of RPCs and the Server rejects/cancels all of the RPCs.

Repro - #23141

### PR DESCRIPTION
**Pull request by yashykt**
Fixes #23146

Add missing reset for ping clocks to avoid mistakenly sending GOAWAY frames due to 'too_many_pings'

### CODE REVIEW

File: test/core/transport/chttp2/too_many_pings_test.cc [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  grpc_metadata_array_destroy(&trailing_metadata_recv);
+  grpc_metadata_array_destroy(&request_metadata_recv);
+  grpc_call_details_destroy(&call_details);
+  grpc_call_unref(c);
+  grpc_call_unref(s);
+  cq_verifier_destroy(cqv);
+  return status;
+}
+
+// Perform a few ALTS handshakes sequentially (using the fake, in-process ALTS
```
Comment (apolcyn) [inline test/core/transport/chttp2/too_many_pings_test.cc:124]: same here about stale comment

File: test/core/transport/chttp2/too_many_pings_test.cc [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+#include "test/core/util/port.h"
+#include "test/core/util/test_config.h"
+
+#include "test/core/end2end/cq_verifier.h"
+
+namespace {
+
+void* tag(int i) { return (void*)static_cast<intptr_t>(i); }
+
+// Perform a simple RPC and capture the value of the authority header
```
Comment (apolcyn) [inline test/core/transport/chttp2/too_many_pings_test.cc:59]: comment nit: this is left over from another test file I originally copy/pasted from :)

File: test/core/transport/chttp2/too_many_pings_test.cc [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  cq_verifier_destroy(cqv);
+  return status;
+}
+
+// Test that sending a lot of RPCs that are cancelled by the server doesn't
+// result in too many pings
+TEST(TooManyPings, TestLotsOfServerCancelledRpcsDoesntGiveTooManyPings) {
+  grpc_completion_queue* cq = grpc_completion_queue_create_for_next(nullptr);
+  // create the server
+  grpc_server* server = grpc_server_create(nullptr, nullptr);
```
Comment (apolcyn) [inline test/core/transport/chttp2/too_many_pings_test.cc:129]: I didn't check myself - are the server keepalive args unnecessary for this to reproduce?

Comment (yashykt) [reply test/core/transport/chttp2/too_many_pings_test.cc:129]: yes, they were unnecessary. The default BDP ping is enough to trigger this.

Comment (yashykt) [reply test/core/transport/chttp2/too_many_pings_test.cc:129]: I should probably document that

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 4]
```diff
@@ -2390,6 +2398,7 @@ static void close_from_api(grpc_chttp2_transport* t, grpc_chttp2_stream* s,
   grpc_slice_buffer_add(&t->qbuf, status_hdr);
   grpc_slice_buffer_add(&t->qbuf, message_pfx);
   grpc_slice_buffer_add(&t->qbuf, grpc_slice_ref_internal(slice));
+  grpc_chttp2_reset_ping_clock(t);
```
Comment (apolcyn) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc:2401]: As I understand the problem here was that the server thought that the client was violating GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS ?

If so I'm a little confused by the documentation around `GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS`, that are described in https://github.com/grpc/grpc/blob/master/doc/keepalive.md.

Mainly this wording: "If there are no data frames being sent on the transport", makes me think that the transport only cares <i>data frames</i>. Should the wording there be expanded beyond data frames to all frame types?

Comment (yashykt) [reply src/core/ext/transport/chttp2/transport/chttp2_transport.cc:2401]: You are right. (Minor correction - "all frame types" - "data/header/window_update"). Created #23151

**Comment by jtattermusch** [comment]
CC @JamesNK

**Pull request merged**